include/hse/hse_experimental.h

```diff
@@ -0,0 +1,147 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ * Copyright (C) 2015-2021 Micron Technology, Inc.  All rights reserved.
+ */
+
+/* The interfaces defined in this file are for internal use only.  The are not
+ * part of the HSE library public API and hence may change in name, arguments,
+ * and/or semantics without notice.
+ */
+#ifndef HSE_EXPERIMENTAL_H
+#define HSE_EXPERIMENTAL_H
+
+#include <hse/flags.h>
+#include <hse/limits.h>
+#include <hse/types.h>
+#include <hse/version.h>
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <limits.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#pragma GCC visibility push(default)
+
+
+/** @brief Request a data compaction operation.
```

Comment:
Look in types.h for doxygen addtogroup directives. You will want to use those here.